### ISSUE TITLE
v0.68.0 Inference bug with optional types

### ISSUE THREAD

**Issue by johnryan** [facebook/flow#6041]
I am running into an issue with flow v0.68.0 as demonstrated here:

https://flow.org/try/#0PTAEAEDMBsHsHcBQBLAtgB1gJwC6hwJ7oCmoA3qAMKwawB2xdOAKkcQDSgCi0xqjOAApZY6AM6gAvqEgjUoAORZiAQwDGOBQG4UtXKABUoFRIBKqjTLmLl6zTsQATYmugrl+NqACyK9AGUcFRxiZlhhUTEAHlN-QQAuUAB5ACMAKxccTn9mAEFmLkTUjI0APlAAXlAACjEgkMSc-K4ASkry2MEdZ1d3UmIAD0x9SABXOg1kelA1egYNKOpURKXMBiZWEiiDUs5OovTM7LyCg5KcUurEUFBUP0Dg0PCRcUTfAPqniPEYuOPm0qIFqJaqzWjrHArGhtCrlVb0ASbYhRAAkABFkJBIFEeHwBN9oktdqBOqVSg5EIQSKACZVyKIcFM6CpoAA1FmjYgAfkSdSwyDoAHNJDpemIJEtQIMQnRHGYLDgAHTwiFRAnlMjXUDKWXELDVNqam43ZQ4UZYOigKKOZAAN1KZBwAAtkGJFegXm6GUyWezoJzJFFgDb7TobpJEBHELM6HVbvdPmFaVVap8YRqtabzZajcbvfRfRziIkAETMYh1EtaiMixDR+hx6h0eYhRx0mMt6p3D6PJOelrVJYtByLOaZYht4DkxBAA

Basically flow is incorrectly inferring the type as `undefined` when then then optional type should be `(undefined | string)`

Related issue in flow-typed: https://github.com/flowtype/flow-typed/issues/1997

**Comment by mrkev** [comment]
Confirming it's still present in 0.70

**Comment by sibelius** [comment]
is this a problem with $Diff?

**Comment by SamChou19815** [comment]
Closing since '*' is removed and that's likely the one causing issue
